src/engine/list.js

```diff
@@ -8,9 +8,7 @@
   * @param {Array} contents Contents of the list, as an array.
   * @constructor
   */
-function List (name, contents) {
+module.exports = function List (name, contents) {
```

Comment:
I don't think this rule is very important since `no-use-before-define` keeps things understandable even with the hoisting from the declaration style.  I think I was going back and forth when I made the changes for `module.exports = function Foo () {}`.
What if it was always consistent e.g.
```javascript
var Foo = function () {
    // ...
};

module.exports = Foo;
```